### ISSUE TITLE
Certain colour combinations results in low contrast text

### ISSUE THREAD

**Issue by alexsmithfanning** [ccrama/Slide#379]
Some colors for custom subreddits colors have the text color incorrect. Especially for cards.

Some examples:

For pure green, the text color is still gray-ish.

For red, the text color is once again still gray-ish.

I can provide many more examples if I need to.

**Comment by ccrama** [comment]
The font color is supplied by the base theme, not the color theme

**Comment by alexsmithfanning** [comment]
Some text is hard to read (really hard to read) when the dark purple color is added to a subreddit. Is it possible to mitigate this by applying font colors to card based on card color?

It seems like the light theme takes a back seat a lot of the time.

**Comment by ccrama** [comment]
I tried that a while back but it ended up slowing the app down bigtime. Looking into it in the future though

**Comment by alexsmithfanning** [comment]
This is a big issue. People using the light theme and actively use subreddit colors will be disappointed.

I think this should be tagged for something in the very near future.

**Comment by alexsmithfanning** [comment]
Actually, a great solution to this would be to disallow colors that cause issues with text. That would work, wouldn't it?

**Comment by ccrama** [comment]
I don't think it's that big of an issue, it has been here since Day 1 and not many people have complained. And that is over half the theme colors so not really an option

**Comment by Alexendoo** [comment]
The problem with light theme is that it in various places displays both white and black text on the main colour, restricting the choices you can take to not see issues in some places quite considerably

**Comment by alexsmithfanning** [comment]
This issue could be easily resolved by restricting what colors the user can choose.

This problem is unavoidable with the current amount of colors to choose from. Its also backwards. The base color chooser shouldn't have a darkness or lightness slider. I has to be a certain shade of a certain color. On the other hand, the accent color can be many different values of a certain color, and text is almost never behind the accent color.

In other words, the color chooser for base colors shouldn't have a tint/shade slider. The accent chooser should have a tint/shade slider instead.

Am I making any sense? I feel like I'm not making any sense...

**Comment by Alexendoo** [comment]
That would require different sets of colours for both themes, and without changing the text colours limiting the colours to only that of mid lightness since dark or light ones won't work properly using the current situation at least in light mode. And I don't particularly want that, I like my dark UI bits since those bits of the UI shouldn't draw attention

**Comment by alexsmithfanning** [comment]
It wouldn't effect the dark theme at all, would it?

**Comment by Alexendoo** [comment]
light theme with the dark main colour would no longer be an option, ie what I have in the screenshot

**Comment by alexsmithfanning** [comment]
You could make it so that only darker colors of the original colors are available on the light theme, and only lighter colors of the original color are available on the dark theme.

Would that work?

**Comment by Alexendoo** [comment]
The dark theme doesn't have issues as much as the light theme, it would be easier to just pick a font colour of say white as it mostly is and only display that colour on top of the main colour, rather than mixing as is done in some situations (mainly in the light theme)

**Comment by alexsmithfanning** [comment]
Okay. Thanks for the input.

**Comment by Alexendoo** [comment]
Fixed with 8833608a010ec71c15acc19dd15b1d093ffd0753
